doc/api/quic.md

```diff
@@ -0,0 +1,2276 @@
+`QuicSession` will automatically update to use the new validated path.
+
+The callback will be invoked with three arguments:
+
+* `result` {string} Either `'failure'` or `'success'`, denoting the status
+  of the path challenge.
+* `local` {Object} The local address component of the tested path.
+* `remote` {Object} The remote address component of the tested path.
+
+The `'pathValidation'` event will be emitted multiple times.
+
+#### Event: `'qlog'`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `jsonChunk` {string} A JSON fragment.
+
+Emitted if the `qlog: true` option was passed to `quicsocket.connect()` or
+`net.createQuicSocket()` functions.
+
+The argument is a JSON fragment according to the [qlog standard][].
+
+The `'qlog'` event will be emitted multiple times.
+
+#### Event: `'secure'`
+<!-- YAML
+added: REPLACEME
+-->
+
+Emitted after the TLS handshake has been completed.
+
+The callback will be invoked with two arguments:
+
+* `servername` {string} The SNI servername requested by the client.
+* `alpnProtocol` {string} The negotiated ALPN protocol.
+* `cipher` {Object} Information about the selected cipher algorithm.
+  * `name` {string} The cipher algorithm name.
+  * `version` {string} The TLS version (currently always `'TLSv1.3'`).
+
+These will also be available using the `quicsession.servername`,
+`quicsession.alpnProtocol`, and `quicsession.cipher` properties.
+
+The `'secure'` event will not be emitted more than once.
+
+#### Event: `'stream'`
+<!-- YAML
+added: REPLACEME
+-->
+
+Emitted when a new `QuicStream` has been initiated by the connected peer.
+
+The `'stream'` event may be emitted multiple times.
+
+#### quicsession.ackDelayRetransmitCount
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the number of retransmissions caused by delayed
+acknowledgements.
+
+#### quicsession.address
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {Object}
+  * `address` {string} The local IPv4 or IPv6 address to which the `QuicSession`
+    is bound.
+  * `family` {string} Either `'IPv4'` or `'IPv6'`.
+  * `port` {number} The local IP port to which the `QuicSocket` is bound.
+
+An object containing the local address information for the `QuicSocket` to which
+the `QuicSession` is currently associated.
+
+#### quicsession.alpnProtocol
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {string}
+
+The ALPN protocol identifier negotiated for this session.
+
+#### quicsession.authenticated
+<!--YAML
+added: REPLACEME
+-->
+* Type: {boolean}
+
+True if the certificate provided by the peer during the TLS 1.3
+handshake has been verified.
+
+#### quicsession.authenticationError
+
+* Type: {Object} An error object
+
+If `quicsession.authenticated` is false, returns an `Error` object
+representing the reason the peer certificate verification failed.
+
+#### quicsession.bidiStreamCount
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bidirectional streams
```

Comment:
No, it's an artifact of how the stats are *currently* recorded (using a struct of uint64_t types). That is going to be revisited with more properly sized types but it's been a low priority. Now that we have `AliasedStruct` as a utility instead of just `AliasedBuffer`, that will be easier to do. For number of streams per `QuicSession`, it will never be greater than max `uint32_t`.